internal/resource/nvml-device.go

```diff
@@ -86,3 +90,37 @@ func (d nvmlDevice) GetTotalMemoryMB() (uint64, error) {
 	}
 	return info.Total / (1024 * 1024), nil
 }
+
+func (d nvmlDevice) GetDisplayMode() (string, error) {
+	info, retVal := d.Device.GetPciInfo()
+	if retVal != nvml.SUCCESS {
+		return "", retVal
+	}
+	var bytes []byte
+	for _, char := range info.BusId {
+		if char == 0 {
+			break
+		}
+		bytes = append(bytes, byte(char))
+	}
+	pciID := strings.ToLower(strings.TrimPrefix(string(bytes), "0000"))
+	return resolvePCIAddressToMode(pciID)
+}
+
+func resolvePCIAddressToMode(addr string) (string, error) {
```

Comment:
This seems like a function that we should have in `go-nvpci` for instead of reimplementing it here.

Reply:
I changed the implementation here to just read the `class`  based on the pci address. Maybe we should still push getting the class function to go-nvml.

Reply:
The place for this to live would always be `go-nvlib`. The package at `go-nvml` is a wrapper for NVML specifically and we don't really add additional functionality there.